packages/lesmis-server/src/app/createApp.ts

```diff
@@ -3,38 +3,22 @@
  * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
  */
 
-import express from 'express';
-import cors from 'cors';
-import bodyParser from 'body-parser';
-import errorHandler from 'api-error-handler';
-import { sessionCookie, attachSessionModel } from '@tupaia/server-boilerplate';
-import { addRoutesToApp } from './addRoutesToApp';
+import { TupaiaDatabase } from '@tupaia/database';
+import { OrchestratorApiBuilder, handleWith } from '@tupaia/server-boilerplate';
 import { LesmisSessionModel } from '../models';
+import { UserRoute, EntityRequest, EntityRoute, EntitiesRoute } from '../routes';
+import { verifyLogin } from '../auth';
 
 /**
  * Set up express server with middleware,
  */
-export function createApp(sessionModel: LesmisSessionModel) {
-  const app = express();
-
-  /**
-   * Add middleware
-   */
-  app.use(
-    cors({
-      origin: true,
-      credentials: true, // withCredentials needs to be set for cookies to save @see https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/withCredentials
-    }),
-  );
-  app.use(bodyParser.json({ limit: '50mb' }));
-  app.use(errorHandler());
-  app.use(sessionCookie());
-  app.use(attachSessionModel(sessionModel));
-
-  /**
-   * Add all routes to the app
-   */
-  addRoutesToApp(app);
-
-  return app;
+export function createApp() {
+  const database = new TupaiaDatabase();
+  return new OrchestratorApiBuilder(database)
+    .useSessionModel(new LesmisSessionModel(database))
+    .verifyLogin(verifyLogin)
+    .get('/v1/user', handleWith(UserRoute))
+    .get('/v1/entities', handleWith(EntitiesRoute))
+    .get<EntityRequest>('/v1/entity/:entityCode', handleWith(EntityRoute))
+    .build();
```

Comment:
@tcaiger I think you might be interested in this. So we no longer use `addRoutesToApp()` and instead we define all the routes here.

Reply:
Hey Rohan, I have had quick look through these changes and it looks awesome! I'm glad to see a backend expert improving the server-boilerplate package.

Reply:
Is my understanding correct that we can do .use() in this chain to extend the functionality of the server when using it? And do you know if it would work to re-use an already used module. ie `.use(bodyParser.json({ limit: '10mb' }))`?

Reply:
Also this is probably a mis-informed question, but do we need to apply the handleWith wrapper here or could that be done in the boilerplate code now?

Reply:
> Is my understanding correct that we can do .use() in this chain to extend the functionality of the server when using it?
Yup, that's correct, `use()` will add a middleware layer to the specified path. As to whether you can overwrite an existing module (ie. bodyParser), I don't think so... If we wanna make this configurable I reckon we ought to break it out of the constructor into its own function, eg.
```
setBodyParserLimit(limit: string) {
  this.app.use(bodyParser.json({ limit }));
  return this;
}
```

> Also this is probably a mis-informed question, but do we need to apply the handleWith wrapper here or could that be done in the boilerplate code now?
Not mis-informed at all, you're totally right that would be ideal 😄. However, we are playing with some pretty complex generics here and I simply was able to find a way to coalesce the method signature of `get/use/etc.` to accept a `Route` class type and then apply the `handleWith` wrapper itself. I'm inclined to go back and give it another shot at some point since it definitely should be possible, it's really just cos my typescript skills are still pretty basic